pytest_valgrind/plugin.py

```diff
@@ -80,15 +82,83 @@ def __init__(self, config):
+            # for objects with cyclic garbage collection support.
+            # Valgrind counts these as "possible" leaks meaning that we would
+            # have a huge amount of false positives (valgrind has hardcoded
+            # heuristics for some cases, but none seem to match Python).
+            # Until a better solution is found, the solution here is to ignore
+            # possible leaks. In the first version I did this by manually
+            # counting leaks, instead of enforcing the use of:
+            #     --errors-for-leak-kinds=definite
+            # So we get the below sanity check: If the user passed the above
+            # command to valgrind, possible leaks are not errors and we can
+            # use the errors as reported by valgrind. If the user did not pass
+            # the above (or similar?) command, we manually count leaks and use
+            # that instead.
+            # The main difference should be that directly counting leaks will
+            # also report indirectly lost memory, while the above command
+            # ignores them. Effectively, we ignore the valgrind setting and
+            # enforce:
+            #     --errors-for-leak-kinds=definite,indirect
+
+            # Disable the GC just in case it might confuse things if it runs
+            # while probing the error reporting behaviour:
+            gc.disable()
+            leaks = do_leak_check()
+            errors = get_valgrind_num_errs()
+            # This will use "interior-pointers" (possible leak):
+            obj = object()
+            new_leaks = do_leak_check() - leaks
+            new_errors = get_valgrind_num_errs() - errors
+            gc.enable()
+            if new_leaks != 0:
+                # There be no new leaks for the above simple code
+                raise RuntimeError(
+                        "Sanity check failed, please report this, as it is "
+                        "probably a pytest-valgrind bug. (leak detected for "
+                        "simple allocation)")
+            if new_errors:
+                self.count_leaks = True
+
+            else:
+                self.count_leaks = False
+                print("pytest-valgrind: Reporting memory errors as set up "
+                      "using the `--errors-for-leak-kinds` option of valgrind.")
+
+        self.log_file_name = config.getvalue("valgrind_log")
+        if self.log_file_name:
+            self.log_file = open(self.log_file_name)
         else:
             self.log_file = None
 
         self.prev_leaked = 0
         self.prev_errors = 0
 
+    def pytest_report_header(self, config):
+        info = []
+        info.append(f"pytest-valgrind: logfile={self.log_file_name}")
+        if self.no_memcheck:
+            info.append("Not performing runtime memory leak checks.")
+
+        elif self.count_leaks:
+            info.append("Reporting direct+indirect leaks and ingoring the")
```

Comment:
You probably meant `ignoring` and not `ingoring`.

Reply:
Thanks, will push a small commit directly to fix this!